cranelift/codegen/src/isle_prelude.rs

```diff
@@ -512,7 +512,15 @@ macro_rules! isle_common_prelude_methods {
         #[inline]
         fn imm64_masked(&mut self, ty: Type, x: u64) -> Imm64 {
             debug_assert!(ty.bits() <= 64);
-            Imm64::new((x & ((1u64 << ty.bits()) - 1)) as i64)
+            if ty.bits() == 64 {
+                Imm64::new(x as i64)
+            } else {
```

Comment:
I don't understand pulling `ty.bits() == 64` out as a special case if you're using the right-shift version below. `u64::MAX >> 0` should be a correct mask for this case.

Reply:
Argh, yeah, I wrote it the conditionalized way first and then decided I liked the right-shift variant better, but only updated locally and didn't actually delete the conditional. I think I've found the rest of my brain now; sorry about that...